pkg/gardenadm/botanist/botanist.go

```diff
@@ -223,12 +239,34 @@ func (b *AutonomousBotanist) initializeFakeGardenResources(ctx context.Context,
 		}
 	}
 
+	return initializeFakeGardenSecrets(ctx, b.GardenClient, secrets, secretBinding, credentialsBinding)
+}
+
+func initializeFakeGardenSecrets(
```

Comment:
I don't like that this is a separate function 😅

Reply:
Joined the functions again.